tech-specs/src/explore/design/ledger/pos-integration.md

```diff
@@ -0,0 +1,257 @@
+# PoS integration
+
+The [PoS system](/explore/design/pos.md) is integrated into Anoma ledger via:
+- an account with an internal address and a [native VP](vp.md#native-vps) and that validates any changes applied by transactions to the PoS account
+- transaction WASMs to perform various PoS actions, also available as a library code for custom made transactions
+
+The `votes_per_token` PoS system parameter must be chosen to satisfy the [Tendermint requirement](https://github.com/tendermint/spec/blob/60395941214439339cc60040944c67893b5f8145/spec/abci/apps.md#validator-updates) of `MaxTotalVotingPower = MaxInt64 / 8`.
+
+All [the data relevant to the PoS system](/explore/design/pos.md#storage) are stored under the PoS account's storage sub-space, with the following key schema (the PoS address prefix is omitted for clarity):
+
+- `parameters`: the system parameters
+- `parameters/{parameters_hash}`: the system parameters value with this `parameters_hash `
+- `parameters/{parameters_hash}/{validator_address}`: a set of validators who want to switch to parameters with this `parameters_hash` (only keys are needed here, so the value can be a `unit`)
+- `validator/{validator_address}/consensus_key`
+- `validator/{validator_address}/state`
+- `validator/{validator_address}/total_deltas`
+- `validator/{validator_address}/voting_power`
+- `slash/{validator_address}`: a list of slashes, where each record contains epoch and slash rate
+- `bond/{bond_source}/{bond_validator}/delta`
+- `unbond/{unbond_source}/{unbond_validator}/deltas`
+- `validator_set/active`
+- `validator_set/inactive`
+- `total_voting_power`
+
+- standard validator metadata (these are regular storage values, not epoched data):
+  - `validator/{validator_address}/staking_reward_address`: an address that should receive staking rewards, if not set, the rewards are sent to `validator_address`
+  - TBA (e.g. alias, website, description, delegation commission rate, etc.)
+
+Additionally, only XAN tokens can be staked. The tokens being staked (bonds and unbonds amounts) are kept in the XAN token account under `{xan_address}/balance/{pos_address}`.
+
+## Initialization
+
+The PoS system is initialized via a native VP interface that is given the validator set for the genesis block.
+
+## Staking rewards and transaction fees
+
+Staking rewards for validators are rewarded in Tendermint's method `BeginBlock` in the base ledger. If a validator has specified `validator/{validator_address}/staking_reward_address` the rewards are credited to this address, otherwise, the validator's address is used.
```

Comment:
Specifying the reward payment address should be mandatory when creating a validator.

Reply:
updated in b72f7c92f03928e95a845342ae8a88cb8dad8458